arviz/tests/test_data.py

```diff
@@ -30,6 +34,60 @@ def chains():
     return 2
 
 
+@pytest.fixture(autouse=True)
+def no_remote_data(tmpdir):
+    """Delete all remote data and replace it with a local dataset."""
+    keys = list(REMOTE_DATASETS)
+    for key in keys:
+        del REMOTE_DATASETS[key]
+
+    centered = LOCAL_DATASETS["centered_eight"]
+    filename = os.path.join(str(tmpdir), os.path.basename(centered.filename))
+    url = "file://" + centered.filename
+    REMOTE_DATASETS["test_remote"] = RemoteFileMetadata(
+        filename=filename,
+        url=url,
+        checksum="9ae00c83654b3f061d32c882ec0a270d10838fa36515ecb162b89a290e014849",
+        description=centered.description,
+    )
+    REMOTE_DATASETS["bad_checksum"] = RemoteFileMetadata(
+        filename=filename, url=url, checksum="bad!", description=centered.description
+    )
+
+
+def test_load_local_arviz_data():
+    assert load_arviz_data("centered_eight")
+
+
+def test_clear_data_home():
+    resource = REMOTE_DATASETS["test_remote"]
+    assert not os.path.exists(resource.filename)
+    load_arviz_data("test_remote")
+    assert os.path.exists(resource.filename)
+    clear_data_home(data_home=os.path.dirname(resource.filename))
+    assert not os.path.exists(resource.filename)
+
+
+def test_load_remote_arviz_data():
+    assert load_arviz_data("test_remote")
+
+
+def test_bad_checksum():
+    with pytest.raises(IOError):
+        load_arviz_data("bad_checksum")
+
+
+def test_missing_dataset():
+    with pytest.raises(ValueError):
+        load_arviz_data("does not exist")
+
+
+def test_list_datasets():
+    datasets = list_datasets()
+    for key in ("centered_eight", "non_centered_eight", "test_remote", "bad_checksum"):
```

Comment:
It might be better to do a set compare. This way if the test fails it'll show all that are missing, not just report the first one it failed on
```suggestion
    for set("centered_eight", "non_centered_eight", "test_remote", "bad_checksum") == set(datasets)
```

Reply:
Sorry, `list_datasets` is just a string representation of all the available datasets. This was making sure that at least the names of the datasets show up in that string. I have commented it a little more to make the intention clear.